### ISSUE TITLE
[Feature]: App Developer can delete their apps with `cf delete APP_NAME` / `DELETE /v3/apps/:guid` endpoint

### ISSUE THREAD

**Issue by tcdowney** [cloudfoundry/cf-k8s-controllers#334]
### Blockers/Dependencies

There's a lot to unpack here, so going to try and break this out into separate stories. Let's consider this one blocked on https://github.com/cloudfoundry/cf-k8s-controllers/issues/335.

The goal of this initial story will be for the CLI command to succeed, the `CFApp` to be deleted, and no `Pods` should be running for it.

### Background

**As an** App Developer
**I want** I want to be able to use `cf delete APP_NAME` to delete my App
**So that** I can remove any trace of my app from the cluster

### Acceptance Criteria

---
### When the app exists
**GIVEN** I have an app deployed and running on the cluster
**WHEN I** run `cf delete APP_NAME`
**THEN I** see the command succeed and I can no longer see the corresponding `CFApp` on the cluster

```bash
$ cf delete dora
Really delete the app dora? [yN]: y
Deleting app dora in org org / space space as admin...
OK
```

---
### When the app no longer exists
**GIVEN** I already deleted my app (or never created it)
**WHEN I** run `cf delete APP_NAME`
**THEN I** see the command succeeds (it tries to be idempotent)

```bash
$ cf delete dora
Really delete the app dora? [yN]: y
Deleting app dora in org org / space space as admin...
OK
```

---
### DELETE /v3/apps/:guid Happy Path

**Request:**
```
curl "https://api.example.org/v3/apps/:app-guid" \
  -X DELETE \
  -H "Authorization: bearer [token]"
```

**Response:**
```
HTTP/1.1 202 Accepted
Location: https://api.example.org/v3/jobs/app.delete-<app-guid>
```

---

### DELETE /v3/apps/:guid App Doesn't Exist
**Request:**
```
curl "https://api.example.org/v3/apps/:non-existant-app-guid" \
  -X DELETE \
  -H "Authorization: bearer [token]"
```

**Response:**
```json
HTTP/1.1 404 Not Found

{
  "errors": [
    {
      "code": 10010,
      "detail": "App not found",
      "title": "CF-ResourceNotFound"
    }
  ]
}
```

---

### Requests made during `cf delete`
```
15:21 $ cf delete dora -f -v | grep -i REQUEST: -A2
REQUEST: [2021-12-07T15:22:34-07:00]
GET /v3/apps?names=dora&space_guids=d260a535-39a2-4a13-b2e2-d262bad4e984 HTTP/1.1
Host: api.plain-tail.capi.land
--
REQUEST: [2021-12-07T15:22:35-07:00]
DELETE /v3/apps/6b237996-ba80-4d4d-b340-9392c398b863 HTTP/1.1
Host: api.plain-tail.capi.land
--
REQUEST: [2021-12-07T15:22:35-07:00]
GET /v3/jobs/a7a341e2-77c7-464d-9296-cb9031b6a6ce HTTP/1.1
Host: api.plain-tail.capi.land
--
REQUEST: [2021-12-07T15:22:38-07:00]
GET /v3/jobs/a7a341e2-77c7-464d-9296-cb9031b6a6ce HTTP/1.1
Host: api.plain-tail.capi.land
--
REQUEST: [2021-12-07T15:22:41-07:00]
GET /v3/jobs/a7a341e2-77c7-464d-9296-cb9031b6a6ce HTTP/1.1
Host: api.plain-tail.capi.land
```

We should have the first call implemented in https://github.com/cloudfoundry/cf-k8s-controllers/pull/270. The following two are new(ish):

* `DELETE /v3/apps/6b237996-ba80-4d4d-b340-9392c398b863`
* `GET /v3/jobs/a7a341e2-77c7-464d-9296-cb9031b6a6ce`

We have a version of `GET /v3/jobs/:guid` that's purposefully built for applying space manifests. We need to figure out how to make that work for this "app deletion" job. We'll do that in a separate story and block this one on it.

Docs: https://v3-apidocs.cloudfoundry.org/version/3.111.0/index.html#delete-an-app
